apollo/src/graphql/storage-provider/storageProviderQueries.ts

```diff
@@ -27,4 +33,85 @@ builder.queryFields((t) => ({
             return storageProvider;
         },
     }),
+    listObjectsForModelVersion: t.field({
+        type: StorageProviderObjectConnection,
+        authScopes: {
+            loggedIn: true,
+        },
+        args: {
+            modelVersionId: t.arg.string({ required: true }),
+            // TODO: Putting defaultValue & required overrides defaultValue
+            first: t.arg({
+                type: 'Limit',
+                defaultValue: 10,
+                required: true,
+            }),
+            after: t.arg.string(),
+            prefix: t.arg.string(),
+        },
+        async resolve(_root, args, ctx) {
+            const modelVersion = (await ObjectionMLModelVersion.query()
+                .findById(args.modelVersionId)
+                .first()) as ObjectionMLModelVersion;
+
+            const parentModel = (await ObjectionMLModel.query()
+                .where('modelId', modelVersion.modelId)
+                .first()) as ObjectionMLModel;
+
+            const project = (await ObjectionProject.query().findById(
+                parentModel.projectId,
+            )) as ObjectionProject;
+            await ObjectionTeamEdge.userHasRole(ctx.user.$id(), project.teamId, [
+                'owner',
+                'member',
+            ]);
```

Comment:
I think this can be simplified by:
```typescript
const modelVersion = (await ObjectionMLModelVersion.query()
    .findById(args.modelVersionId)
    .first()) as ObjectionMLModelVersion;
const team = (await ObjectionMLModel
    .relatedQuery('getTeam')
    .for(modelVersion.modelId).first()) as ObjectionTeam;

await ObjectionTeamEdge.userHasRole(
    ctx.user.$id(),
    team.$id(),
    ['owner', 'member', 'viewer']
);
```

Reply:
The check is to avoid data exhilaration. In general, anyone with any degree of team membership (i.e., an edge exists relating the user and team) is able to view all objects associated with that team. We don't want someone who is not a member of the team to have any permissions to view or modify a team's objects.

I don't think I totally understand the bit about invoking the query outside of the web client or CLI. Auth checks should never exist in the client because a client application can always be modified. If auth checks are always server side, then mode of access doesn't matter

Reply:
Oh yeah, duh.

For the second bit, I was referring to someone just trying to hit our API on their own. But nevermind that point is moot :)

Reply:
`ObjectionMLModel.relationMappings.getTeam: Property join.through is not supported for this relation type.` I see that this relation is defined, so that's weird.